src/app/modules/product/services/product-api.service.ts

```diff
@@ -28,6 +28,12 @@ export class ProductService {
       .pipe(map((response: any) => new Product(response.payload)));
   }
 
+  public updateProduct(product: Product) {
```

Comment:
Please add type to this function. 
`public updateProduct(product: Product) : Observable<Product>`